app/src/main/java/edu/drexel/lapcounter/lapcounter/frontend/WorkoutDetailsActivity.java

```diff
@@ -1,60 +1,97 @@
 package edu.drexel.lapcounter.lapcounter.frontend;
 
 import android.arch.lifecycle.ViewModelProviders;
+import android.content.Intent;
 import android.os.Bundle;
 import android.support.v7.app.AppCompatActivity;
 import android.widget.TextView;
 
 import java.text.DateFormat;
 import java.text.SimpleDateFormat;
+import java.util.ArrayList;
 import java.util.Date;
 import java.util.TimeZone;
 
 import edu.drexel.lapcounter.lapcounter.R;
+import edu.drexel.lapcounter.lapcounter.backend.Database.Device.Device;
+import edu.drexel.lapcounter.lapcounter.backend.Database.Device.DeviceViewModel;
 import edu.drexel.lapcounter.lapcounter.backend.TimestampConverter;
 import edu.drexel.lapcounter.lapcounter.backend.Database.Workout.WorkoutViewModel;
-import edu.drexel.lapcounter.lapcounter.backend.Database.Workout.Workouts;
+import edu.drexel.lapcounter.lapcounter.backend.Database.Workout.Workout;
 import edu.drexel.lapcounter.lapcounter.frontend.navigationbar.NavBar;
 
 public class WorkoutDetailsActivity extends AppCompatActivity {
 
 
     private final NavBar mNavBar = new NavBar(this);
     private WorkoutViewModel mWorkoutViewModel;
+    private DeviceViewModel mDeviceViewModel;
+    private int mWorkout_id;
     private final static int MILLISECONDS_IN_SECOND = 1000;
 
+    private static String qualify(String s) {
+        return WorkoutDetailsActivity.class.getPackage().getName() + "." + s;
+    }
+    public static final String EXTRAS_WORKOUT_ID = qualify("DEVICE_ADDRESS");
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_workout_details);
 
-        //In production version, this ID (or startdate) will be passed from the
-        //past work/workout history s
-        int id = 10;
+        Intent intent = getIntent();
+        mWorkout_id = intent.getIntExtra(EXTRAS_WORKOUT_ID,0);
+
         // In the final version, use R.string.<string id> for titles
         getSupportActionBar().setTitle("Workout Details");
 
         mNavBar.init();
 
         mWorkoutViewModel = ViewModelProviders.of(this).get(WorkoutViewModel.class);
-        Workouts workout = new Workouts();
+        Workout workout = new Workout();
+
+        mDeviceViewModel = ViewModelProviders.of(this).get(DeviceViewModel.class);
+        Device d = new Device("Test_Device","00:11:00:33:AA",-25);
+
+
 
+        mDeviceViewModel.insert(d);
+        //This code maintains a known entry in the DB for testing, the insert will update the
+        //workout with ID 10 in the DB.
         workout.setID(10);
```

Comment:
Ah, is this why you needed the on-conflict logic in the DAO? If that's helpful for testing, that's okay for now, but remember to remove it once things are integrated.